### PR TITLE
Feature/arcii model 2.0 dev

### PR DESCRIPTION
**Pull request by pl8787**

### CODE REVIEW

File: matchzoo/layers/matching_layer.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+"""An implementation of Matching Layer."""
+import typing
+
+from keras import backend as K
+from keras.engine import Layer
+
+
+class MatchingLayer(Layer):
```
Comment (bwanglzu) [inline matchzoo/layers/matching_layer.py]: can you refine the class name?

File: matchzoo/layers/matching_layer.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            is the cosine proximity between the two samples.
+        :param matching_type: the similarity function for matching
+        :param **kwargs: Standard layer keyword arguments.
+        """
+        super(MatchingLayer, self).__init__(**kwargs)
+        self.normalize = normalize
+        self.matching_type = matching_type
+        self.supports_masking = True
+        self.valid_matching_type = ['dot', 'mul', 'plus', 'minus', 'concat']
+        if matching_type not in self.valid_matching_type:
```
Comment (bwanglzu) [inline matchzoo/layers/matching_layer.py]:
```suggestion
        if matching_type not in ['dot', 'mul', ...]:
```

File: matchzoo/layers/matching_layer.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        """
+        Build the layer.
+
+        :param input_shape: the shapes of the input tensors,
+            for MatchingLayer we need tow input tensors.
+        """
+        # Used purely for shape validation.
+        if not isinstance(input_shape, list) or len(input_shape) != 2:
+            raise ValueError('A `MatchingLayer` layer should be called '
+                             'on a list of 2 inputs.')
```
Comment (bwanglzu) [inline matchzoo/layers/matching_layer.py]: 'a list of 2 inputs' .& 'input type should be a list'

File: matchzoo/layers/matching_layer.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    def compute_output_shape(self, input_shape: list) -> tuple:
+        """
+        Calculate the layer output shape.
+
+        :param input_shape: the shapes of the input tensors,
+            for MatchingLayer we need tow input tensors.
+        """
+        if not isinstance(input_shape, list) or len(input_shape) != 2:
+            raise ValueError('A `MatchingLayer` layer should be called '
+                             'on a list of 2 inputs.')
```
Comment (bwanglzu) [inline matchzoo/layers/matching_layer.py]: same

File: matchzoo/models/arcii_model.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    def _conv_pool_block(
+        self,
+        input: typing.Any,
+        kernel_count: int,
+        kernel_size: int,
+        padding: str,
+        activation: str,
+        pool_size: int
+    ) -> typing.Any:
+        output = Conv2D(kernel_count,
```
Comment (bwanglzu) [inline matchzoo/models/arcii_model.py]: can you refine the variable name?

File: matchzoo/layers/matching_layer.py (hunk 3 above)
Comment (pl8787) [reply matchzoo/layers/matching_layer.py]: 'a list of 2 inputs' contains 'input type of list'

File: matchzoo/models/arcii_model.py [hunk 6]
```diff
@@ -17,6 +17,13 @@ class ArcIIModel(engine.BaseModel):
         >>> import matchzoo as mz
```
Comment (faneshion) [inline matchzoo/models/arcii_model.py]: Remove these two lines as the `train_pack` has not been used.

**Pull request merged**